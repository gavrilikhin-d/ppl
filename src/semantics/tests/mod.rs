use crate::test_compilation_result;

test_compilation_result!(candidate_not_viable);
test_compilation_result!(constraints);
test_compilation_result!(constraints_in_constructor);
test_compilation_result!(generics);
test_compilation_result!(missing_fields);
test_compilation_result!(multiple_errors);
test_compilation_result!(multiple_initialization);
test_compilation_result!(non_class_constructor);
test_compilation_result!(predeclare_vars);
test_compilation_result!(references);
test_compilation_result!(traits);
test_compilation_result!(type_as_value);
test_compilation_result!(wrong_initializer_type);
