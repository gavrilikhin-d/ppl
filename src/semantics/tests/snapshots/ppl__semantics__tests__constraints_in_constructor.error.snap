---
source: src/semantics/tests/mod.rs
expression: err
---
Error: semantics::type_mismatch

  × expected `T` type, got `Rational`

Error:   × T
   ╭─[constraints_in_constructor.ppl:2:5]
 1 │ type Point<T>:
 2 │     x, y: T
   ·        ┬
   ·        ╰── this has `T` type
 3 │ 
   ╰────
Error:   × Rational
   ╭─[constraints_in_constructor.ppl:4:18]
 3 │ 
 4 │ Point { x: 1, y: 0.1 }
   ·                  ─┬─
   ·                   ╰── this has `Rational` type
   ╰────
